src/devices/fakeBattery/fakeBattery.cpp

```diff
@@ -134,7 +116,60 @@ bool FakeBattery::getBatteryInfo(string& info)
     return true;
 }
 
-void FakeBattery::threadRelease()
+void FakeBattery::setBatteryVoltage(const double voltage)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_voltage = voltage;
+    updateStatus();
+}
+
+void FakeBattery::setBatteryCurrent(const double current)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_current = current;
+    updateStatus();
+}
+
+void FakeBattery::setBatteryCharge(const double charge)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_charge = charge;
+    updateStatus();
+}
+
+void FakeBattery::setBatteryStatus(const fakeBatteryStatus status)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_status = static_cast<yarp::dev::IBattery::Battery_status>(status);
+    updateStatus();
+}
+
+void FakeBattery::setBatteryInfo(const std::string& info)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_info = info;
+}
+
+void FakeBattery::setBatteryTemperature(const double temperature)
 {
-    yTrace("FakeBattery Thread released\n");
+    std::lock_guard<std::mutex> lock(m_mutex);
+    battery_temperature = temperature;
+}
+
+void FakeBattery::updateStatus()
```

Comment:
I wonder if this whole method could be put in a base class, so other battery devices (e.g. bcbBattery, bmsBattery) can reuse it. (Maybe in the next PR....)